test/amp/amp_test.go

```diff
@@ -0,0 +1,271 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build !windows
+
+package amp
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"path/filepath"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	sigv4 "github.com/aws/aws-sdk-go-v2/aws/signer/v4"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/environment"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric/dimension"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/test_runner"
+	"github.com/aws/amazon-cloudwatch-agent-test/util/common"
+)
+
+type AMPResponse struct {
+	Status string
+	Data   AMPResponseData
+}
+type AMPResponseData struct {
+	ResultType string
+	Result     []AMPDataResult
+}
+type AMPDataResult struct {
+	Metric map[string]interface{}
+	Value  []interface{}
+}
+
+const (
+	namespace = "AMPDestinationTest"
+	// template prometheus query for getting average of 3 min
+	ampQueryTemplate = "avg_over_time(%s%s[3m])"
```

Comment:
Using PromQL function `avg_over_time` for last 3m https://prometheus.io/docs/prometheus/latest/querying/functions/#aggregation_over_time